src/workerd/io/worker.c++

```diff
@@ -1652,6 +1652,11 @@ Worker::Worker(kj::Own<const Script> scriptParam,
             // evaluation is complete.
             TmpDirStoreScope tmpDirStoreScope;
 
+            // We allow eval and new Function() on bootstrap phase, and later disable
+            // them to prevent malicious code from executing.
```

Comment:
Nitpick: I don't think we've ever used the term "bootstrap phase" for this before. We just call it "startup" or "global scope execution".

Also the reason we disallow eval is a little more subtle than "malicious code".

```suggestion
            // We allow eval and new Function() during startup, becaues startup time is entirely
            // deterministic, so we can easily reproduce the input to eval() by just running the
            // worker again. We do not allow eval() at runtime because we need to have a record of
            // all code that executes in production for forensic purposes, and at runtime the input
            // to eval() could have come from a remote source on which we don't have a record.
```

Reply:
Thanks you're right!